fsspec/caching.py

```diff
@@ -597,7 +597,7 @@ def __init__(
 
             self.data = dict(zip(offsets, blocks))
         else:
-            self.data = data
+            self.data = {}
```

Comment:
```suggestion
            self.data = dict()
```
`{}` is an empty set, not an empty dict. I'm surprised mypy didn't throw a fit. 

Technically behavior change since the empty dict is copy instead of pass by reference, but should be fine I suppose.

Reply:
I could change to:
```python
        else if data is None:
            self.data = dict()
        else:
            self.data = data
```
However, I believe the most common idiom is to handle `None` and empty containers the same: create a new empty container.

Reply:
> {} is an empty set, not an empty dict

This is not so

```python
>>> type({})
dict
```

This is because `{}` was around in the language before sets existed at all (dicts were there from the start)

Reply:
ruff says:
```
fsspec/caching.py:600:25: C408 Unnecessary `dict` call (rewrite as a literal)
```

Reply:
Whoops, got them mixed up. 👍

Reply:
Perhaps rule C408 should be disabled.

Reply:
Indeed `dict` would be consistent with `dict(zip(offsets, blocks))`, two lines above.